multiaccounts/settings_wallet/database.go

```diff
@@ -0,0 +1,153 @@
+package walletsettings
+
+import (
+	"database/sql"
+	"errors"
+)
+
+type TokenPreferences struct {
+	Key           string `json:"key"`
+	Position      int    `json:"position"`
+	GroupPosition int    `json:"groupPosition"`
+	Visible       bool   `json:"visible"`
+	CommunityID   string `json:"communityId"`
+}
+
+type TokenPreferencesData struct {
+	Preferences      []TokenPreferences `json:"preferences"`
+	Testnet          bool               `json:"testnet"`
+	GroupByCommunity bool               `json:"groupByCommunity"`
+}
+
+type WalletSettings struct {
+	db *sql.DB
+}
+
+func NewWalletSettings(db *sql.DB) *WalletSettings {
+	return &WalletSettings{
+		db: db,
+	}
+}
+
+// This function should not be used directly, it is called from the functions which update token preferences.
+func (db *WalletSettings) setClockOfLastTokenPreferencesChange(tx *sql.Tx, clock uint64) error {
+	if tx == nil {
+		return nil
+	}
+	_, err := tx.Exec("UPDATE settings SET wallet_token_preferences_change_clock = ? WHERE synthetic_id = 'id'", clock)
+	return err
+}
+
+func (db *WalletSettings) GetClockOfLastTokenPreferencesChange() (result uint64, err error) {
+	query := "SELECT wallet_token_preferences_change_clock FROM settings WHERE synthetic_id = 'id'"
+	err = db.db.QueryRow(query).Scan(&result)
+	if err != nil {
+		return 0, err
+	}
+	return result, err
+}
+
+func (db *WalletSettings) setTokenGroupByCommunity(tx *sql.Tx, value bool) error {
+	if tx == nil {
+		return nil
+	}
+	_, err := tx.Exec("UPDATE settings SET wallet_token_preferences_group_by_community = ? WHERE synthetic_id = 'id'", value)
+	return err
+}
+
+func (db *WalletSettings) GetTokenGroupByCommunity() (result bool, err error) {
+	query := "SELECT wallet_token_preferences_group_by_community FROM settings WHERE synthetic_id = 'id'"
+	err = db.db.QueryRow(query).Scan(&result)
+	if err != nil {
+		return false, err
+	}
+	return result, err
+}
+
+func (db *WalletSettings) UpdateTokenPreferences(preferences []TokenPreferences, groupByCommunity bool, testNetworksEnabled bool, clock uint64) error {
+	if len(preferences) == 0 {
+		return errors.New("tokens: trying to create custom order with empty list")
+	}
+
+	tx, err := db.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	var mainError error = nil
+
+	defer func() {
+		if mainError == nil {
+			err = tx.Commit()
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	_, err = tx.Exec("DELETE FROM token_preferences WHERE testnet = ?", testNetworksEnabled)
```

Comment:
Why not use the `mainError` directly? There is no overwrite here. Same question also for other places.